rules.tf

```diff
@@ -0,0 +1,968 @@
+locals {
+  byte_match_statement_rules = module.this.enabled && var.byte_match_statement_rules != null ? {
+    for indx, rule in flatten(var.byte_match_statement_rules) :
+    format("%s-%s",
+      lookup(rule, "name", null) != null ? rule.name : format("%s-%d", module.this.id, indx),
```

Comment:
if `rule.name` is not defined in the variables, this 

```
format("%s-%s",
      lookup(rule, "name", null) != null ? rule.name : format("%s-%d", module.this.id, indx),
rule.action
```

can create rules with the same names (for the same action).
Is it ok?

Or should we add the category name to the final rule name, similar to

```
format("%s-byte-match-%d", module.this.id, indx)
```